src/basic/assert-util.c

```diff
@@ -17,6 +18,22 @@ void log_set_assert_return_is_critical(bool b) {
         assert_return_is_critical = b;
 }
 
+void log_set_assert_return_is_critical_from_env(void) {
+        static int cached = INT_MIN;
+        int r;
+
+        if (cached == INT_MIN) {
+                r = getenv_bool("SYSTEMD_ASSERT_RETURN_IS_CRITICAL");
```

Comment:
this really should be secure_getenv_bool

Reply:
For some reasons, `secure_getenv_bool()` does not work...